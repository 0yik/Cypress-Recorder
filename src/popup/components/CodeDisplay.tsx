import * as React from 'react';

export interface CodeDisplayProps { }

export const CodeDisplay: React.FC = (props: CodeDisplayProps) => (
  <div>
    CodeDisplay
  </div>
);
