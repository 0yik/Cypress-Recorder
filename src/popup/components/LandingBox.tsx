import * as React from 'react';

export interface LandingBoxProps { }

export const LandingBox = (props: LandingBoxProps) => (
  <div>
    
  </div>
);
