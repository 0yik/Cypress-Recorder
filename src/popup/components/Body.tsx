import * as React from 'react';
import { RecState } from '../../types';

import CodeDisplay from './CodeDisplay';
import LandingBox from './LandingBox';

export interface BodyProps {
  isValidTab: boolean,
  recStatus: RecState,
  codeBlocks: string[],
}

export default ({ recStatus, codeBlocks, isValidTab }: BodyProps) => (
  <div id="body">
    {recStatus === 'off' && <LandingBox isValidTab={isValidTab} />}
    {recStatus !== 'off' && <CodeDisplay codeBlocks={codeBlocks} />}
  </div>
);
