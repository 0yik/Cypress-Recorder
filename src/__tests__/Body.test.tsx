import * as React from 'react';
import { shallow } from 'enzyme';
import Body from '../popup/components/Body';
import LandingBox from '../popup/components/LandingBox';
import CodeDisplay from '../popup/components/CodeDisplay';

import '../setupTests';

describe('Body', () => {
  let wrapper;
  let props;
  beforeAll(() => {
    props = {
      codeBlocks: [],
      isValidTab: true,
      destroyBlock: jest.fn(),
      moveBlock: jest.fn(),
    };
  });
  it('Should render LandingBox when recStatus is off', () => {
    props.recStatus = 'off';
    wrapper = shallow(<Body {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.exists(LandingBox)).toBe(true);
    expect(wrapper.exists(CodeDisplay)).toBe(false);
  });
  it('Should render CodeDisplay when recStatus is paused', () => {
    props.recStatus = 'paused';
    wrapper = shallow(<Body {...props} />);
    expect(wrapper.exists(CodeDisplay)).toBe(true);
    expect(wrapper.exists(LandingBox)).toBe(false);
  });
  it('Should render CodeDisplay when recStatus is on', () => {
    props.recStatus = 'on';
    wrapper = shallow(<Body {...props} />);
    expect(wrapper.exists(CodeDisplay)).toBe(true);
    expect(wrapper.exists(LandingBox)).toBe(false);
  });
});
