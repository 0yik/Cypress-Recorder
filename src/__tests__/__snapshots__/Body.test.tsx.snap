// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Should render CodeDisplay when recStatus is on 1`] = `
<div
  id="body"
>
  <Component
    codeBlocks={Array []}
    destroyBlock={[MockFunction]}
    moveBlock={[MockFunction]}
  />
</div>
`;

exports[`Body Should render CodeDisplay when recStatus is paused 1`] = `
<div
  id="body"
>
  <Component
    codeBlocks={Array []}
    destroyBlock={[MockFunction]}
    moveBlock={[MockFunction]}
  />
</div>
`;

exports[`Body Should render LandingBox when recStatus is off 1`] = `
<div
  id="body"
>
  <Component
    isValidTab={true}
  />
</div>
`;
